### PR TITLE
Feat logging

### PR DESCRIPTION
**Pull request by mcmcgrath13**
* Use `logging` functionality instead of prints
* Default is still to print to stdout
* Added `logging` params section to `outputs` params, which can instead send the logs to file (timestamped) or set a different log level (though currently most logs are `INFO`)
* Retry stochastic integration tests a few times before declaring failure

### CODE REVIEW

File: titan/params/outputs.yml [hunk 1]
```diff
@@ -36,3 +36,21 @@ outputs:
       - drug_types
       - locations
       - components
+  logging:
+    to_file:
```
Comment (s-bessey) [inline titan/params/outputs.yml:40]: Would it make more sense to put this as `stream` with file and stdout?

File: titan/params/outputs.yml [hunk 2]
```diff
@@ -36,3 +36,21 @@ outputs:
...
+      type: boolean
+      description: Whether to log to a file instead of the console (default)
+    filepath:
+      default: '__cwd__'
+      description: The directory path wherethe logs will be printed to if `to_file` is true.  The logs will be named `titan_<run id>_log.txt`.  The default is your current working directory.
+      type: any
+    level:
+      default: 'INFO'
+      type: enum
+      description: What is the minimum log level to record?
```
Comment (s-bessey) [inline titan/params/outputs.yml:51]: I'm not sure it's self-explanatory what these levels indicate. Can that be explained on the webapp?

File: titan/params/outputs.yml (hunk 1 above)
Comment (mcmcgrath13) [reply titan/params/outputs.yml:40]: `destination`?

File: titan/params/outputs.yml (hunk 2 above)
Comment (mcmcgrath13) [reply titan/params/outputs.yml:51]: we could probably link out to a generic description somewhere

File: titan/params/outputs.yml (hunk 1 above)
Comment (s-bessey) [reply titan/params/outputs.yml:40]: That sounds good to me!

File: titan/params/outputs.yml (hunk 2 above)
Comment (s-bessey) [reply titan/params/outputs.yml:51]: Yeah, that sounds like a good plan

File: titan/params/outputs.yml [hunk 3]
```diff
@@ -37,18 +37,21 @@ outputs:
...
+      default: stdout
+      type: enum
+      description: Whether to log to a file or the console (default)
+      values:
+        - stdout
+        - file
     filepath:
       default: '__cwd__'
-      description: The directory path wherethe logs will be printed to if `to_file` is true.  The logs will be named `titan_<run id>_log.txt`.  The default is your current working directory.
+      description: The directory path wherethe logs will be printed to if `destination` is `file`.  The logs will be named `titan_log_<date/time>.txt`.  The default is your current working directory.
```
Comment (s-bessey) [inline titan/params/outputs.yml:49]: Typo on "where the" 🤕

**Review by s-bessey — APPROVED** [review]

**Pull request merged**